RentACar/RentACar/Repositories/Repository.cs

```diff
@@ -0,0 +1,64 @@
+﻿using RentACar.Helpers;
+using RentACar.Intefaces;
+using RentACar.Models.Base;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace RentACar.Repositories
+{
+    public abstract class Repository<T> : IRepository<T> where T : Entity
+    {
+        private List<T> _objs;
+
+        private readonly string _jsonFile;
+
+        public Repository(string config)
+        {
+            _jsonFile = config;
+            _objs = CrudHelper<T>.CheckFileAndGetList(_jsonFile);
+        }
+
+        public virtual T Create(T obj)
+        {
+            obj.Id = CrudHelper<T>.GetNewId(_objs);
+            _objs.Add(obj);
+
+            CrudHelper<T>.SaveListToFile(_objs, _jsonFile);
+
+            return obj;
+        }
+
+        public virtual T Update(T obj)
+        {
+            var existing = Get(obj.Id);
+
+            if (existing is null)
+                return null;
+
+            UpdateObject(existing, obj);
+
+            CrudHelper<T>.SaveListToFile(_objs, _jsonFile);
```

Comment:
the more I see this static helper the more I hate it, hopefully we will drop it soon hahaha

Reply:
AHhahaha.. sorry for you! :P